uast/transformer/ops.go

```diff
@@ -458,7 +476,17 @@ func JoinObj(ops ...ObjectOp) ObjectOp {
 	if partial != nil {
 		required = nil
 	}
-	return opObjJoin{ops: out, partial: partial, allFields: required}
+	for i := 0; i < len(out); i++ {
+		op := out[i]
+		if len(op.fields) != 0 {
+			continue
+		}
+		if o, ok := op.op.(Obj); ok && len(o) == 0 && len(out) > 1 {
+			out = append(out[:i], out[i+1:]...)
```

Comment:
This could be potentially slow depending on how the runtime preallocates the slice.

Reply:
This happens only once on the program startup when the transforms are being built. So not so relevant.